src/cockpit/bridge.py

```diff
@@ -62,10 +63,15 @@ def __init__(self, args: argparse.Namespace):
         self.superuser_rule = SuperuserRoutingRule(self, args.privileged)
         self.internal_bus.export('/superuser', self.superuser_rule)
 
+        # we run this after the channel rules, so we don't need to check source="internal"
+        self.pcp_rule = PeerRoutingRule(self, ['/usr/libexec/cockpit-pcp'], {},
+                                        lambda msg: msg.get('payload') == 'metrics')
+
         super().__init__([
             HostRoutingRule(self),
             self.superuser_rule,
             ChannelRoutingRule(self, CHANNEL_TYPES),
+            self.pcp_rule,
```

Comment:
So to be clear: This does not use the routing rule in `pkg/pcp/manifest.json` at all, but instead hardcodes it. With the C bridge you could add third-party bridges without changing cockpit code, which was quite cool (I don't know how much this is actually being used in practice, though).

This already hardcodes the superuser rules, which breaks our recent feature to make them more flexible (PR #17536), on [customer demand](https://bugzilla.redhat.com/show_bug.cgi?id=2091034). But the machinery that we now have in place here seems powerful enough that it could just actually read the `match` rules and the `spawn` program vector from the manifest?

Reply:
To clarify: I don't mind hardcoding this here as a provisional step towards building the bridge. I'd just like to discuss the eventual plan, i.e. that this is a temporary hack until we implement manifest reading.